simplelru/lru.go

```diff
@@ -50,11 +71,28 @@ func (c *LRU[K, V]) Purge() {
 func (c *LRU[K, V]) Add(key K, value V) (evicted bool) {
 	// Check for existing item
 	if ent, ok := c.items[key]; ok {
-		c.evictList.MoveToFront(ent)
+		if c.useSieve {
+			ent.Visited = true
+		} else {
+			c.evictList.MoveToFront(ent)
+		}
+
 		ent.Value = value
 		return false
 	}
 
+	if c.useSieve {
+		if c.evictList.Length() >= c.size {
+			c.performSieveEviction()
+			evicted = true
+		}
+
+		ent := c.evictList.PushFront(key, value)
+		ent.Visited = false
+		c.items[key] = ent
+		return
```

Comment:
do we need to return true here? it would be easier to understand if we can put the lru eviction in else part

Reply:
It is not required, since we are using named returns.  https://go.dev/tour/basics/7

>In the Go libraries, you'll find that when an if statement doesn't flow into the next statement—that is, the body ends in break, continue, goto, or return—the unnecessary else is omitted.
[Link](https://go.dev/doc/effective_go#control-structures)

Reply:
got it! I have not used go for a long time. Haha, happy to learn again. Thank you! :)